js/entities.js

```diff
@@ -1,94 +1,273 @@
+        // accomodate Entity(String) and Entity(Object) or Entity(JSONString)
+        if(typeof entity === "string") {
+            try {
+                entity = JSON.parse(entity)
+            } catch(e) {
+                entity = { id: entity }
+            }
+        }
+        const id = entity.id ?? entity["@id"] ?? entity // id is primary key
+        if(!id) { throw new Error("Entity must have an id") }
+        if(EntityMap.has(id)) { throw new Error("Entity already exists")}
+        this.Annotations = new Map()
+        this.data = entity
+    }
+    
+    get assertions() {
+        let clone = JSON.parse(JSON.stringify(this.data))
+        this.Annotations.forEach(annotation => applyAssertions(clone, annotation.normalized))
+        this._assertions = clone
+        return this._assertions
+    }
+    
+    get data() {
+        return this._data
+    }
+    
+    get id() {
+        return this.data.id
+    }
+    
+    set data(entity) {
+        entity.id = entity.id ?? entity["@id"] ?? entity // id is primary key
+        if(objectMatch(this._data, entity)) {
+            console.warn("Entity data unchanged")
+            return
+        }
+        const oldRecord = this._data ? JSON.parse(JSON.stringify(this._data)) : {}
+        this._data = entity
+        EntityMap.set(this.id, this)
+        this.#announceUpdate()
+        if(!objectMatch(oldRecord.id, this.id)) { this.#resolveURI(true).then(this.#findAssertions).then(this.#announceNewEntity) }
+    }
+
+    attachAnnotation(annotation) {
+        this.Annotations.set(annotation.id, annotation)
+    }
+
+    #findAssertions = (assertions) => {
+        var annos = Array.isArray(assertions) ? Promise.resolve(assertions) : findByTargetId(this.id,[],DEER.URLS.QUERY)
+        return annos
+            .then(annotations => annotations.filter(a=>(a.type ?? a['@type'])?.includes("Annotation")).map(anno => new Annotation(anno)))
+            .then(newAssertions => newAssertions?.length ? this.#announceUpdate() : this.#announceComplete())
+            .catch(err => console.log(err))
+    }
+
+    #resolveURI = (withAssertions) =>{
+        const targetStyle = ["target", "target.@id", "target.id"]
+        let historyWildcard = { "$exists": true, "$size": 0 }
+        let obj = { "$or": [{'@id': this.id}], "__rerum.history.next": historyWildcard }
+        for (let target of targetStyle) {
+            let o = {}
+            o[target] = this.id
+            obj["$or"].push(o)
+        }
+        var results = Boolean(withAssertions) ? fetch(DEER.URLS.QUERY,{
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify(obj)
+        }) : fetch(this.id)
+        return results
+        .then(res => res.ok ? res.json() : Promise.reject(res))
+        .then(finds => {
+            if(finds.length === 0) { return Promise.reject({status:404}) }
+            this.data = finds?.find(e => e['@id'] === this.id) ?? finds
```

Comment:
The annotations don't seem to make it through here.

Reply:
#103 issue